server/repository/src/db_diesel/program_requisition/program_supplier.rs

```diff
@@ -0,0 +1,314 @@
+            ..Default::default()
+        };
+
+        let program2 = ProgramRow {
+            id: "program2".to_string(),
+            master_list_id: master_list2.id.clone(),
+            ..Default::default()
+        };
+
+        let master_list_name_join1 = MasterListNameJoinRow {
+            id: "master_list_name_join1".to_string(),
+            name_id: name1.id.clone(),
+            master_list_id: master_list1.id.clone(),
+        };
+        let master_list_name_join2 = MasterListNameJoinRow {
+            id: "master_list_name_join2".to_string(),
+            name_id: store_name1.id.clone(),
+            master_list_id: master_list1.id.clone(),
+        };
+        let master_list_name_join3 = MasterListNameJoinRow {
+            id: "master_list_name_join3".to_string(),
+            name_id: store_name2.id.clone(),
+            master_list_id: master_list2.id.clone(),
+        };
+
+        let name_store_join1 = NameStoreJoinRow {
+            id: "name_store_join1".to_string(),
+            name_id: name1.id.clone(),
+            store_id: store3.id.clone(),
+            ..Default::default()
+        };
+        let name_store_join2 = NameStoreJoinRow {
+            id: "name_store_join2".to_string(),
+            name_id: store_name1.id.clone(),
+            store_id: store3.id.clone(),
+            ..Default::default()
+        };
+
+        let name_store_join3 = NameStoreJoinRow {
+            id: "name_store_join3".to_string(),
+            name_id: store_name2.id.clone(),
+            store_id: store3.id.clone(),
+            ..Default::default()
+        };
+
+        let (_, connection, _, _) = setup_all_with_data(
+            "program_supplier_repository",
+            MockDataInserts::none(),
+            MockData {
+                stores: vec![store1.clone(), store2.clone(), store3.clone()],
+                names: vec![
+                    name1,
+                    store_name1.clone(),
+                    store_name2.clone(),
+                    store_name3.clone(),
+                ],
+                master_lists: vec![master_list1, master_list2],
+                programs: vec![program1.clone(), program2.clone()],
+                master_list_name_joins: vec![master_list_name_join1, master_list_name_join3],
+                name_store_joins: vec![name_store_join1.clone(), name_store_join2.clone()],
+                ..Default::default()
+            },
+        )
+        .await;
+
+        // TEST 1 without master list join for store 1 and without name_store_join for store 2
+        // should result in nothing (since name1 is not store)
+        let repo = ProgramSupplierRepository::new(&connection);
+
+        let filter = ProgramSupplierFilter::new()
+            .program_id(EqualFilter::equal_any(vec![
+                program1.id.clone(),
+                program2.id.clone(),
+            ]))
+            .name(NameFilter::new().is_store(true).is_visible(true));
```

Comment:
Am I right in understanding that these 2 filters should always be set when querying program_suppliers?
Shouldn't we default them on or hardcode the filters even if it potentially creates a bit of duplication in the resulting sql query? Is there a use case for this endpoint to show non-visible names for a store?

Reply:
Thanks, this was done based on previous comments, there was no need to expose them